### ISSUE TITLE
False positive with `**kwargs` with multiple arguments

### ISSUE THREAD

**Issue by eltoder** [python/mypy#18759]
**To Reproduce**

```python
def foo(x: int, y: str) -> None:
    return

def bar() -> None:
    args = {"x": 123, "y": "abc"}
    foo(**args)
```
https://mypy-play.net/?mypy=latest&python=3.12&gist=97467e43c0e316f2f9fd1d6f0c2bd432

**Expected Behavior**

This should type check without errors. It's fairly common and while not 100% safe, checking the type of `args` is not enough anyway -- you have to check the _values_ of the keys. This information is not present in types.

**Actual Behavior**

This generates these errors:
```
main.py:6: error: Argument 1 to "foo" has incompatible type "**dict[str, object]"; expected "int"  [arg-type]
main.py:6: error: Argument 1 to "foo" has incompatible type "**dict[str, object]"; expected "str"  [arg-type]
```
Note that two contradictory errors are generated on the same line.

**Your Environment**

- Mypy version used: 1.15.0
- Python version used: 3.11, 3.12

**Comment by A5rocks** [comment]
I swear there was an earlier issue about the same thing but I can't find anything...

This is an issue with the error message IMO. mypy is correct and shouldn't let this typecheck. If you want gradual typing then annotate `args` accordingly. `args: dict[str, Any]` should work just fine.

**Comment by eltoder** [comment]
Of course I can work-around any issue by throwing Any at things, but this is still a false positive for a completely valid code. False positives are very bad, because they teach people to silence errors without thinking. 

The correct thing to do, which is not hard, is to check that the value type in the dict is a subset of the union of the types of all arguments not passed explicitly.

**Comment by A5rocks** [comment]
> this is still a false positive for a completely valid code

So is something like `x: int = ["a", 5][1]` in a similar fashion. (IMO dict-as-kwargs shouldn't be accepted except as args to a `**kwargs` argument but whatever, I'm probably too strict)

**Comment by asottile-sentry** [comment]
```python
class _Kwargs(TypedDict):
    x: int
    y: str

def foo(x: int, y: str) -> None:
    return

def bar() -> None:
    args: _Kwargs = {"x": 123, "y": "abc"}
    foo(**args)
```

I recently highlighted this pattern [in a video of mine](https://www.youtube.com/watch?v=Nb2mIrfUHNw)

**Comment by sterliakov** [comment]
This looks exactly like #5382 (and related #11583) - mypy does not infer TypedDict types from dict literals.
